packages/wonder-blocks-core/components/provider-group.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import * as React from "react";
+
+type Provider<P> = {|
+    type: React.ComponentType<P>,
+    props?: P,
```

Comment:
Question: Should we consider catering to providers that need the value of an earlier provider to be bound to one of their props?

If so, what syntax might that take? An additional optional value here like:

```
type Binding = {|
    propName: string,
    bindingPath: string,
|};

type Provider<P> = {|
...
    bindings?: Array<Binding>,
|}
```

Where binding path is of the form:

```
providerIndex.argumentIndex.fieldName1.fieldName2
```

Where `providerIndex` is the provider index in the `providers` prop, `argumentIndex` is the index of the argument from that provider's children function call, and each `fieldNameX` is a dereference level in that specific argument object.

So an examples might be:

```
0.0
```

or 

```
0.0.get("test")
```

Deconstructing the string, we could look in the `provided` array at index 0, then grab the argument at index 0 from that specific subarray, then call the function with the arguments and bind that to the named prop.

   `provided[0][0]["get"]("test")`

Of course, if we never envisage a need for this, we can completely leave it out, but it seems like it might be useful to build in.

Reply:
The `Binding` feels complicated, maybe a callback the signature of the `children` function that would've been passed to the CaaF if it was being called normally.

Reply:
I'm not sure how your suggestion would work, @kevinbarabash. This is for if a given CaaF component has a prop that needs to be bound to the value of another CaaF in the list. Not sure how matching the signature of the `children` function achieves that.